homeassistant/components/recorder/__init__.py

```diff
@@ -623,6 +583,122 @@ def run(self, instance: Recorder) -> None:
 KEEP_ALIVE_TASK = KeepAliveTask()
 
 
+@dataclass
+class _RecorderRunsHistory:
+    """Bisectable history of RecorderRuns."""
+
+    run_timestamps: list[int]
+    runs_by_timestamp: dict[int, RecorderRuns]
+
+
+class RunHistory:
+    """Track recorder run history."""
+
+    def __init__(self) -> None:
+        """Track recorder run history."""
+        self.recording_start: datetime = dt_util.utcnow()
```

Comment:
This is moved code but I'm not sure this is a good pattern

I think we should set this in async_initialize or similar instead as we sometimes re init if the SQLite database is detected as corrupt

Reply:
Maybe this should be protected as well?  I don't think we need to ever access it externally except in a test

Reply:
The Sqlite corruption recovery test should also validate that the current run changes